src/App.tsx

```diff
@@ -15,48 +17,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+export enum SortType {
+  Alphabet,
+  Length,
+  Default,
+}
+
+function getSortedGoods(
+  goods: string[],
+  sortField: SortType,
+  isReversed: boolean,
+): string[] {
+  const sortedGoodsCopy = [...goods];
+
+  if (sortField === SortType.Alphabet) {
+    sortedGoodsCopy.sort((good1, good2) => {
+      return good1.localeCompare(good2);
+    });
+  }
+
+  if (sortField === SortType.Length) {
+    sortedGoodsCopy.sort((good1, good2) => {
+      return good1.length - good2.length;
+    });
+  }
+
+  if (isReversed) {
+    sortedGoodsCopy.reverse();
+  }
+
+  return sortedGoodsCopy;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.Default);
+  const [isReversed, setIsReversed] = useState(false);
+  const sortedGoods = getSortedGoods(goodsFromServer, sortField, isReversed);
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn(
+            'button',
+            'is-info',
+            { 'is-light': sortField !== SortType.Alphabet },
+          )}
+          onClick={() => {
+            setSortField(SortType.Alphabet);
+          }}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn(
+            'button',
+            'is-success',
+            { 'is-light': sortField !== SortType.Length },
+          )}
+          onClick={() => {
+            setSortField(SortType.Length);
+          }}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn(
+            'button',
+            'is-warning',
+            { 'is-light': !isReversed },
+          )}
+          onClick={() => {
+            setIsReversed(!isReversed);
+          }}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortField !== SortType.Default || isReversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortField(SortType.Default);
+              setIsReversed(false);
```

Comment:
will be better create separated function for this logic like `reset`

Reply:
done)